src/actions/actions.js

```diff
@@ -411,6 +412,126 @@ export function markCellNotRendered() {
   }
 }
 
+function loginSuccess(userData) {
+  return (dispatch) => {
+    dispatch({
+      type: 'LOGIN_SUCCESS',
+      userData,
+    })
+    dispatch(updateAppMessages('You are logged in'))
+  }
+}
+
+function loginFailure() {
+  return (dispatch) => {
+    dispatch(updateAppMessages('Login Failed'))
+  }
+}
+
+export function login() {
+  const url = '/auth/github'
+  const name = 'github_login'
+  const specs = 'width=500,height=600'
+  const authWindow = window.open(url, name, specs)
+  authWindow.focus()
+
+  return (dispatch) => {
+    // Functions to be called by child window
+    window.loginSuccess = (userData) => {
+      dispatch(loginSuccess(userData))
+    }
+    window.loginFailure = () => dispatch(loginFailure())
+  }
+}
+
+export function logout() {
+  return (dispatch) => {
+    fetch('/logout')
+      .then(response => response.json())
+      .then((json) => {
+        if (json.status === 'success') {
+          dispatch({ type: 'LOGOUT' })
+          dispatch(updateAppMessages('Logged Out'))
+        } else dispatch(updateAppMessages('Logout Failed'))
+      })
+  }
+}
+
+export function exportGist() {
+  // Go through all gist of the user
+  // Match the discription and fileName for each gist
+  // If mactch found, update it
+  // If none found then create a new Gist
+  const API_ROUTE = 'https://api.github.com'
+  return (dispatch, getState) => {
+    const state = getState()
+    const filename = titleToHtmlFilename(state.title)
+    let matchDescription
+    let gistExist = false
+    const gistData = {
+      description: state.title,
+      public: true,
+      files: {
+        [filename]: { content: exportJsmdBundle(state) },
+      },
+    };
+    fetch(`${API_ROUTE}/gists?access_token=${state.userData.accessToken}`)
+      .then(response => response.json())
+      .then((json) => {
+        matchDescription = json.filter(gist =>
+          gist.description === gistData.description &&
```

Comment:
Longer term I am not sure if we want to title the gists and update, but it's probably what we want for now (until we have some kind of server-side saving of state).